pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -1008,6 +1008,12 @@ func (r *Builder) ensureVolumePopulatorPVC(workload *model.Workload, image *mode
 			originalVolumeDiskId = imageProperty.(string)
 		}
 
+		// Check if there's a storage map available
+		if len(r.Context.Map.Storage.Spec.Map) == 0 {
+			err = liberr.New("no storage map found in the migration plan")
```

Comment:
this doesn't necessarily mean that there's no storage map, there could be an empty storage map, no?
I guess the proper way to handle it would be using a validation on the storage map but if that's complicated, how about replacing the code that takes an arbitrary storage class from the storage map with an arbitrary storage class from the inventory?

Reply:
I can try to ensure there is an actual destination available, but I don't we should pick an arbitrary one from the inventory because in the current approach there's an actual way (albeit slightly hacky) for the user ensure the image ends up on a storage class they pick rather than a random one

Reply:
yeah, maybe I was carried away by this suggestion - there's not much difference between picking an arbitrary storage class from the storage map or from the source environments but I agree that the latter is not better. so maybe we should just pick the default storage class for virtualization?..

Reply:
and maybe I jumped too quickly to the solution - my concern with this approach of raising an error here (rather than when validating the storage map) is that I don't see how it would work from the UI: I create a plan with a single VM that is based on an image, so when I get to the network mapping dialog, the UI won't suggest any volume type to map so I'll proceed with an empty mapping and we'll get to this code, right?

Reply:
currently UI suggests all volume types so I'm not entirely sure how it would look

Reply:
> currently UI suggests all volume types so I'm not entirely sure how it would look

specifically for OpenStack?..

Reply:
ok, as we spoke offline - it's ok to keep this code when the UI would specify a "dummy" volume type for images in OpenStack